Samples/PushNotifications/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,205 @@
+
+        // Caller's responsibility to keep the channel alive
+        co_return channel;
+    }
+    else if (result.Status() == PushNotificationChannelStatus::CompletedFailure)
+    {
+        std::cout << result.ExtendedError() << "We hit a critical non-retryable error with channel request!" << std::endl;
+        co_return nullptr;
+    }
+    else
+    {
+        LOG_HR_MSG(result.ExtendedError(), "Some other failure occurred.");
+        co_return nullptr;
+    }
+
+};
+
+winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel RequestChannel()
+{
+    auto task = RequestChannelAsync();
+    if (task.wait_for(std::chrono::seconds(300)) != AsyncStatus::Completed)
+    {
+        task.Cancel();
+        return nullptr;
+    }
+
+    auto result = task.GetResults();
+    return result;
+}
+
+// Register Push Event for Foreground
+void RegisterForegroundNotificationsHandler(const winrt::Microsoft::Windows::PushNotifications::PushNotificationChannel& channel)
+{
+    winrt::event_token token = channel.PushReceived([](auto const&, PushNotificationReceivedEventArgs const& args)
+        {
+            auto payload = args.Payload();
+
+            // Do stuff to process the raw payload
+            std::string payloadString(payload.begin(), payload.end());
+            std::cout << "Push notification content received from FOREGROUND: " << payloadString << std::endl;
+
+            // Set handled to true to prevent background activation
+            args.Handled(true);
+        });
+}
+
+int main()
+{
+    // An unpackaged app must initialize dynamic dependencies so we can consume the Windows App SDK APIs.
+    HRESULT loadWindowsAppSDKHr = LoadWindowsAppSDK();
+    if (FAILED(loadWindowsAppSDKHr))
+    {
+        std::wcout << "Could not load Windows App SDK!" << std::endl;
+        return 1;
+    }
+
+    PushNotificationActivationInfo info(PushNotificationRegistrationActivators::ProtocolActivator);
+    PushNotificationManager::RegisterActivator(info);
+    // We do not call PushNotificationManager::UnregisterActivator
+    // because then we wouldn't be able to receive background activations, once the app has closed.
+
+    auto args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    auto kind = args.Kind();
+    switch (kind)
+    {
+
+    // When it is launched normally (by the users, or from the debugger), the sample requests a Channel Uri and displays it, then waits for notifications.
+    // This user can take a copy of the Channel Uri and use it to send notifications to the sample
+    case ExtendedActivationKind::Launch:
+    {
+
+        // Request a channel which can be passed off to an external app to send notifications to.
+        // The channel uniquely identifies, this app for this user and device.
+        PushNotificationChannel channel = RequestChannel();
+
+        // register the channel, so we can receive notifications in the foreground while the app is running.
```

Comment:
We don't register the channel. We subscribe to an event which is signaled when foreground notifications arrive.